File: Assignment8/program2.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program2.c
//  Description : Accept a Integer and returns Area of Rectangle
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 17/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : RectArea
//  Description   : Returns Area of Rectangle for input float
//  Input         : Float
//  Output        : Double
//
////////////////////////////////////////////////////////////////

double RectArea(float fWidth,float fHeight)
{
     double Area = fWidth*fHeight;
     return Area;
}

int main()
{
    float fValue1 = 0.0, fValue2 = 0.0;
    double dRet = 0.0;
    
    printf("Enter Width");
    scanf("%f",&fValue1);
    
    printf("Enter height");
    scanf("%f",&fValue2);
    
    dRet=RectArea(fValue1,fValue2);
    
    printf("Area of rectangle : %lf",dRet);
    
    return  0;
}

File: Assignment26/program2.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program2.java
//  Description : Accept a String and a Character and returns freqency of that character in String
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : CountChar
//  Description   : Inputs String and a Charater and returns freqency of that character in String
//  Input         : String
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

class Display
{
    public int CountChar(char[] chVal, int iSize, char chSearch)
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if(chVal[iCnt] == chSearch)
            {
                iCount++;
            }
        }
        return iCount;
    }
}
public class program2
{
    public static void main(String args[])
    {
        int iCnt = 0;
        int iLength = 0;
        char[] cValue = null;
        String sValue = null;
        char cSearch = '\0';
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter String : ");
        sValue = sobj.nextLine();

        System.out.println("Enter the character you wanna Search in String : ");
        cSearch = sobj.next().charAt(0);

        iLength = sValue.length();

        cValue = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            cValue[iCnt] = sValue.charAt(iCnt);
        }

        Display dobj = new Display();

        iRet = dobj.CountChar(cValue, iLength, cSearch);

        System.out.println("Frequency of "+cSearch+" in our String is : "+iRet);
        
        sobj.close();
    }
}